### PR TITLE
Add licenses

### LINKED ISSUE #121: nixos-homepage has no license
Someone was talking about sending patches to this repo, but isn't able to unless a license is clearly specified, due to their work.

### PR DESCRIPTION
**Pull request by davidak**
This repository should have an open source license!

The suggestion is to use:

- CC-BY-SA for website content (including artwork)
- MIT for source code (same as nixpkgs)

Next steps:

1. First we need to find a consent for licenses to use. Please provide feedback!
2. To license the contributions, we need every contributor to accept the license. If someone don't accept or we can't reach them, we have to rewrite the contribution!

License accepted:

- [ ] follower
- [x] sanmai-NL
- [x] ghuntley
- [ ] ekalosak
- [x] Phreedom
- [x] ret
- [x] third3ye
- [x] 3noch
- [x] 8573
- [x] aelsabbahy
- [x] andir
- [x] aneeshusa
- [x] aschmolck
- [x] aszlig
- [x] benley
- [x] Blaisorblade
- [x] c0bw3b
- [x] caugner
- [x] civodul
- [x] cko
- [x] cx405
- [X] davidak
- [x] domenkozar
- [x] edolstra
- [x] eqyiel
- [x] ericsagnes
- [x] fpletz
- [x] globin
- [x] goodwillcoding
- [x] grahamc
- [x] gurjeet
- [x] hoodunit
- [x] hyperfekt
- [x] jameysharp
- [x] justgage
- [x] kojiromike
- [x] leenaars
- [x] loskutov
- [x] magnetophon
- [x] manveru
- [x] matthewbauer
- [x] matthiasbeyer
- [x] Mic92
- [x] nckx
- [x] nh2
- [x] noffle
- [x] peti
- [x] phunehehe
- [x] psibi
- [x] qknight
- [x] raboof
- [x] ragnard
- [x] rbvermaa
- [x] rdpate
- [x] ryantm
- [x] samueldr
- [x] shlevy
- [x] statianzo
- [x] sternmull
- [x] svanderburg
- [x] tilpner
- [x] timbertson
- [x] ToxicFrog
- [x] tvon
- [x] unode
- [x] vcunat
- [x] wmertens
- [x] woffs
- [x] zimbatm
- [x] Zimmi48
- [x] zraexy

Close #121

### CODE REVIEW

**Comment by grahamc** [comment]
CC-BY-SA isn't really appropriate for software (like the build scripts), so we may need a collection of licenses:

 - a license for the build scripts
 - a license for the content
 - a license for the logo and imagery

also, it isn't exactly clear that we want people to be running their own nixos.org website.

I've asked @armijnhemel what he thinks and he sighed and said it isn't straight-forwardly easy.

**Comment by armijnhemel** [comment]
I agree with @grahamc here. Creative commons should not be used for software (as build scripts can be regarded as software). It would be OK for artwork.

https://creativecommons.org/faq/#can-i-apply-a-creative-commons-license-to-software

**Comment by armijnhemel** [comment]
For content, such as the homepage or documentation, CC would, of course, be OK.

**Comment by samueldr** [comment]
The logo and imagery may have to be handled here:

 * https://github.com/NixOS/nixos-artwork

The logo itself was figured out:

 * https://github.com/NixOS/nixos-artwork/tree/master/logo

@mogorman would have to confirm for the releases artwork, it somehow escaped me [in my attempt](https://github.com/NixOS/nixos-artwork/issues/23) at wrangling up attribution and licenses.

 * https://github.com/NixOS/nixos-artwork/tree/master/releases

**Comment by grahamc** [comment]
I'd propose we apply the following licenses:

 - CC-BY-SA specifically limited to the *content* of the website (Armijn confirmed this makes sense)
 - MIT for the *software*

**Comment by armijnhemel** [comment]
If you are using existing Javascript libraries you need to look at the license of the Javascript as well. I hope you like fishing, as I just opened a can of worms ;-)

**Comment by mogorman** [comment]
I am totally fine with any license changes project puts forward @samueldr The logos I've made thus far for the project have been combinations of our logo and public domain artwork, so we are able to license it however we wish.

**Comment by grahamc** [comment]
OK let's get screenshots of that post and save the email from @mogorgman saying it ;)

**Comment by primeos** [comment]
> Not sure if all previous contributors have to agree changing the license.

@davidak I'm not a lawyer (would be great if we had one for this), but AFAIK this is indeed required if you want to do it right (though there might be a chance that trivial contributions can be excluded). It is probably also possible to revert changes (if someone doesn't agree) or exclude certain files (but IIRC I haven't actually read anything about this).

I've also wondered about this I while ago and found the following site quite useful: https://opensource.guide/legal/#what-if-i-want-to-change-the-license-of-my-project

> Also, if you neglected to license your project from the start, adding a license is effectively the same as changing licenses.

> Your project’s existing copyright holders. If you’re the sole contributor to your project then either you or your company is the project’s sole copyright holder. You can add or change to whatever license you or your company wants to. Otherwise there may be other copyright holders that you need agreement from in order to change licenses. Who are they? People who have commits in your project is a good place to start. But in some cases copyright will be held by those people’s employers. In some cases people will have only made minimal contributions, but there’s no hard and fast rule that contributions under some number of lines of code are not subject to copyright. What to do? It depends. For a relatively small and young project, it may be feasible to get all existing contributors to agree to a license change in an issue or pull request. For large and long-lived projects, you may have to seek out many contributors and even their heirs. Mozilla took years (2001-2006) to relicense Firefox, Thunderbird, and related software.

Currently the following applies:
https://opensource.guide/legal/#are-public-github-projects-open-source
> Making your GitHub project public is not the same as licensing your project. Public projects are covered by GitHub’s Terms of Service, which allows others to view and fork your project, but your work otherwise comes with no permissions.
>
> If you want others to use, distribute, modify, or contribute back to your project, you need to include an open source license. For example, someone cannot legally use any part of your GitHub project in their code, even if it’s public, unless you explicitly give them the right to do so.

That's basically all I know, hope it helps.
Thanks for pushing this, would be awesome to have a license :)

**Comment by armijnhemel** [comment]
adding @silverhook who is a lawyer

**Comment by davidak** [comment]
@primeos yes, that confirms what i have heard.

We need every contributor to accept the new license or rewrite their contributions. Practical speaking, we have to tag or contact them and they have to say "I license my contributions under the MIT license." We can do that in this PR.

That can be a huge task including a lot of discussions, like you can see here: https://github.com/netdata/netdata/pull/3695 (They created a for-profit company to hold the copyright and created a CLA so the company can change the license)

(So always add a LICENSE when you create a new repositories!)

**Comment by armijnhemel** [comment]
@silverhook said he'd look into this after FOSDEM.

**Comment by davidak** [comment]
Another question for a lawyer and the community: What should be the license holder?

Something like: "The nixos-homepage contributors"?

**Comment by vcunat** [comment]
In nixpkgs we have
> Copyright (c) 2003-2018 Eelco Dolstra and the Nixpkgs/NixOS contributors

I suppose for simplicity we might go with a similar formulation here.  The homepage won't have so many contributors, so it wouldn't be so difficult to get all the attributions, but i don't expect "we" will want to bother with legal action against some copyright infringement of our web anyway, so what would be the point...

**Comment by davidak** [comment]
>The homepage won't have so many contributors, so it wouldn't be so difficult to get all the attributions

There are 65 contributors, way too much to put in one line. And i don't think it's needed since the information is in git. It would get out of sync quickly.

**Comment by Ekleog** [comment]
You can have contributors split into multiple lines. Example: https://git.gnupg.org/cgi-bin/gitweb.cgi?p=gnupg.git;a=blob;f=AUTHORS;h=d5d78143812fae51b0917e0a7ee8c2d8eacf8635;hb=HEAD

**Comment by silverhook** [comment]
Hey, sorry for the late show-up.

Yeah, in general you need approval for re-licensing from all of the previous authors/copyright holders. The only way you could get away with not having to, is if the previous license gave you all the rights to change the license to your new license – e.g. from MIT to GPL-3.0; or from CC-BY to CC-BY-SA (or had a  CLA, or even CAA in place, where they agreed to re-licensing in advance). Even then questions might arise if the changes were big enough to warrant the new work and therefore license.

In any case, if this repo had no license (or CLA) under which the contributors contributed so far, all its content should be deemed as under “all rights reserved” by their authors/copyright holders (+ GitHub ToS). So you *need* their agreement to the license “change”.

MIT is fine for code, CC is fine for other content (images, website texts, etc.) …as long as the rights are all where they need be.

Regarding the copyright holder, there are many way how to do it. The “contributors to” plus their names (and e-mail addresses/websites if corporations) in `CONTRIBUTORS` file is my preferred method, as it keeps the headers short, yet still includes all the names, if needed. The problem is that it (explicitly) assumes that the contributors listed are also copyright holders. E.g. if someone works on NixOS in their company time, it is very likely that it is actually their employer who holds the copyright. So from the copyright PoV, the company should be listed. That is another can of worms on how to handle copyright holders vs authors.

The `year-year` range and its annual forced bumping is a pet peeve of mine. Personally, I strongly suggest to have in the copyright header just the year of the file creation and then leave it. If anyone is worried about this, see my comment here: https://gitlab.com/tildes/tildes/issues/161#note_89229746

Regarding REUSE – great that you want to apply it. Can warmly recommend! But a 3.0 is in the works that aims at simplifying its use and adoption quite a bit (incl. with MIT/BSD licenses). That being said, @davidak as you recently looked into it, you are very welcome to get in touch with (or join) the REUSE team and provide feedback. I think it would be very welcome and valuable.

**Comment by davidak** [comment]
@silverhook thanks for the clarification!

What do you think about the proposed license statement in this PR? Is it clear what "website content" and "code" is or do we have to state the license for every single file?

File: README.md [hunk 1]
```diff
@@ -11,3 +11,11 @@ build it:
     [nix-shell]$ python2 -m SimpleHTTPServer 8000
 
 then open http://127.0.0.1:8000/index.html
+
+## License
+
+The content of the website is licensed under the [Creative Commons Attribution Share Alike 4.0 International](LICENSES/CC-BY-SA-4.0.txt) license.
+
+The software is licensed under the [MIT](LICENSES/MIT.txt) license.
```
Comment (silverhook) [inline README.md]:
```suggestion
The software (including sample code) is licensed under the [MIT](LICENSES/MIT.txt) license.
```

**Comment by silverhook** [comment]
@davidak, it looks clear enough, but it might make sense to explicitly cover also sample code – see my comment: https://github.com/NixOS/nixos-homepage/pull/260/files#r268771018

In general it is a good idea to have license and copyright information in every file (and it is also a specific requirement of REUSE). This is not a legal requirement (although some licenses might ask for it), but is a best practice – e.g if a just a file from the repo gets copied, how is one to know which license it was (and still is) under? That is why it makes sense to have them everywhere. Again, not a hard requirement, but a very very good idea.

**Comment by davidak** [comment]
In the next step, every former contributor has to state:

> I hereby license my contributions to this repository under the following licenses:
> 
> Creative Commons Attribution Share Alike 4.0 International for website content
> MIT for the software (including sample code)

or comment if you see any problem with that.

@gurjeet, @tilpner, @hyperfekt, @sternmull, @ekalosak, @c0bw3b, @andir, @magnetophon, @shlevy, @eqyiel, @woffs, @vcunat, @peti, @cx405, @matthiasbeyer, @loskutov, @rdpate, @unode, @benley, @Mic92, @kojiromike, @ToxicFrog, @nh2, @leenaars, @Zimmi48, @hoodunit, @statianzo, @manveru, @3noch, @cko, @Phreedom, @aneeshusa, @matthewbauer, @ericsagnes

**Comment by davidak** [comment]
ping @aelsabbahy, @aschmolck, @tvon, @phunehehe, @ret, @psibi, @justgage, @ragnard, @domenkozar, @goodwillcoding, @wmertens, @third3ye, @davidak, @caugner, @fpletz, @zraexy, @Blaisorblade, @8573, @raboof, @globin, @nckx, @qknight, @civodul, @noffle, @timbertson, @aszlig, @ryantm, @jameysharp, @svanderburg, @zimbatm, @samueldr, @grahamc, @rbvermaa, @edolstra

**Comment by davidak** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by sternmull** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by Blaisorblade** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by aelsabbahy** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by tvon** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by svanderburg** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by peti** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by psibi** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by tilpner** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by ryantm** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by woffs** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by matthiasbeyer** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by hyperfekt** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by unode** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by samueldr** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by loskutov** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by benley** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by nckx** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by ragnard** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by vcunat** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by caugner** [comment]
I hereby license my contributions to this repository under the following licenses:
Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by statianzo** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by hoodunit** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by aschmolck** [comment]
I hereby license my contributions to this repository under the following licenses: Creative Commons Attribution Share Alike 4.0 International for website content MIT for the software (including sample code)
--

**Comment by magnetophon** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by timbertson** [comment]
I only contributed the logo, which as @samueldr mentioned is already
sorted. But for the sake of completeness:

I hereby license my contributions to this repository under the following
licenses:

Creative Commons Attribution Share Alike 4.0 International for website
content
MIT for the software (including sample code)

>

**Comment by jameysharp** [comment]
Okay, sure...

I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by noffle** [comment]
Heck yes!

---

I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by nh2** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by grahamc** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by rdpate** [comment]
I hereby dedicate my contributions to this repository to the public domain.

(Was [surprised](https://github.com/NixOS/nixos-homepage/commit/bdcb710e08135767a3f5fd27820c27b18c116814) to get the notice.)

**Comment by manveru** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by domenkozar** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by Zimmi48** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content MIT for the software (including sample code)

**Comment by zimbatm** [comment]
I hereby license my contributions to this repository under the following licenses: Creative Commons Attribution Share Alike 4.0 International for website content MIT for the software (including sample code)

**Comment by phunehehe** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)
Or any OSI approved licenses

**Comment by civodul** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
“MIT” for the software (including sample code).

By “MIT” I actually mean “the Expat license”, which I believe is also
what you meant: <http://directory.fsf.org/wiki/License:Expat> (it’s the
one used by Nixpkgs.)

Ludo’.

**Comment by andir** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by leenaars** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
Creative Commons Zero 4.0 International (or later) for website content
MIT for any software (including sample code)
All OSI and FSF recognised licenses for any software (including sample code)

**Comment by leenaars** [comment]
Actually, it is probably better to use the SPDX codes:
 
Content: CC0-1.0,  CC-BY-NC-SA-4.0
Code: MIT-0

**Comment by armijnhemel** [comment]
> Actually, it is probably better to use the SPDX codes:
> 
> Content: CC0-1.0, CC-BY-NC-SA-4.0

Please note: CC-BY-NC-SA is not the same as CC-BY-SA and @silverhook suggested CC-BY-SA.

**Comment by grahamc** [comment]
I'm concerned that these comments are making the approvals we've received so far not valid. @leenaars are you okay with CC-BY-SA?

**Comment by silverhook** [comment]
> Actually, it is probably better to use the SPDX codes:

Yes, very much agree regarding [SPDX IDs](https://spdx.org/ids)

> Content: CC0-1.0, CC-BY-NC-SA-4.0
> Code: MIT-0

As @armijnhemel mentioned, the `NC` (non-commercial) bit might be problematic, as it is non-FOSS. CC-BY-SA-4.0 should be fine though.

**Comment by davidak** [comment]
@silverhook i used the SPDX Full name since it's more human readable.

**Comment by aszlig** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by leenaars** [comment]
I hereby license my contributions to this repository under the following licenses:

Content: CC0-1.0, CC-BY-SA-4.0
Code: MIT-0

**Comment by ToxicFrog** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by gurjeet** [comment]
It took me some time to get an approval from my employer. With their permission I can now say that...

I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by davidak** [comment]
@gurjeet thanks for your effort!

**Comment by edolstra** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content 

MIT for the software (including sample code)

**Comment by zraexy** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by 3noch** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by globin** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content

MIT for the software (including sample code)

**Comment by c0bw3b** [comment]
I hereby license my contributions to this repository under the following licenses:

- Creative Commons Attribution Share Alike 4.0 International (`SPDX-License-Identifier: CC-BY-SA-4.0`) for website content
- MIT (`SPDX-License-Identifier: MIT`) for the software, including sample code

**Comment by wmertens** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by goodwillcoding** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by kojiromike** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by cx405** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 (or any later) International for website content
MIT for the software (including sample code)

**Comment by Mic92** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 (or any later) International for website content
MIT for the software (including sample code)

**Comment by qknight** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by aneeshusa** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by matthewbauer** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content MIT for the software (including sample code)

**Comment by grahamc** [comment]
Here are the contributions we haven't received relicensing approval for yet:

- [ ] https://github.com/NixOS/nixos-homepage/commits?author=8573
- [ ] https://github.com/NixOS/nixos-homepage/commits?author=cko
- [ ] https://github.com/NixOS/nixos-homepage/commits?author=ekalosak
- [ ] https://github.com/NixOS/nixos-homepage/commits?author=eqyiel
- [ ] https://github.com/NixOS/nixos-homepage/commits?author=ericsagnes
- [ ] https://github.com/NixOS/nixos-homepage/commits?author=fpletz
- [ ] https://github.com/NixOS/nixos-homepage/commits?author=justgage
- [ ] https://github.com/NixOS/nixos-homepage/commits?author=Phreedom
- [ ] https://github.com/NixOS/nixos-homepage/commits?author=rbvermaa
- [ ] https://github.com/NixOS/nixos-homepage/commits?author=ret
- [ ] https://github.com/NixOS/nixos-homepage/commits?author=shlevy
- [ ] https://github.com/NixOS/nixos-homepage/commits?author=third3ye

**Comment by davidak** [comment]
@grahamc @silverhook can we accept @leenaars latest comment and just ignore CC0?

**Comment by vcunat** [comment]
For content he specified two licenses, meaning anyone can choose which license to use (for those changes).  I expect it's OK.  The MIT* distinction shouldn't apply, as the only commit changed a SVG file (i.e. not code).

**Comment by davidak** [comment]
@vcunat so the PR has both licenses but we choose just CC-BY-SA. That sounds ok.

**Comment by fpletz** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content MIT for the software (including sample code)

**Comment by silverhook** [comment]
>  can we accept @leenaars latest comment and just ignore CC0?

Which one are you referring to? The way I see it, he agreed with CC0-1.0.

**Comment by davidak** [comment]
@silverhook the second one https://github.com/NixOS/nixos-homepage/pull/260#issuecomment-483970286

**Comment by silverhook** [comment]
> @silverhook the second one [#260 (comment)](https://github.com/NixOS/nixos-homepage/pull/260#issuecomment-483970286)

Ah, that. Best to ask @leenaars. But even if you need just CC-BY-SA-4.0, CC0-1.0 more than allows you to use contributions in a CC-BY-SA-4.0 as well.

**Comment by davidak** [comment]
@silverhook ok, then we can accept it.

**Comment by shlevy** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by justgage** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content MIT for the software (including sample code)

**Comment by davidak** [comment]
@8573, @cko, @ekalosak, @eqyiel, @ericsagnes, @Phreedom, @ret, @third3ye

Please accept the licenses as described in https://github.com/NixOS/nixos-homepage/pull/260#issuecomment-483310698!

If someone could contact them, it would be very helpful to inform them about this issue.

**Comment by ericsagnes** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by eqyiel** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by cko** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by 8573** [comment]
I hereby license my contributions to this repository under the following licenses:
- Creative Commons Attribution Share Alike 4.0 International for website content
- MIT for the software (including sample code)

**Comment by davidak** [comment]
Since i see no way to contact @third3ye, i would go the way to recreate his contribution. It's luckily just removing an "s" (https://github.com/NixOS/nixos-homepage/commit/7e686cd5df9f0dd051816eaed81d53189c2c86e5).

@silverhook what are the steps to do that in a legal way? Just revert his commit and make a new commit that we created on our own? Is it OK that his commit is still in the git history (with the old license)?

**Comment by edolstra** [comment]
That change falls below the https://en.wikipedia.org/wiki/Threshold_of_originality so I think we can proceed without approval for that commit..

**Comment by armijnhemel** [comment]
I agree with @edolstra here

**Comment by davidak** [comment]
@edolstra i agree about the actual patch, but what about the commit description? I'm not sure.

**Comment by raboof** [comment]
We're not trying to relicense the commit description, so that seems fine?

IANAL though, so let's await word from the actual experts here :)

**Comment by davidak** [comment]
Same applies for this small contribution:

- https://github.com/NixOS/nixos-homepage/commit/e960765a95098ae1fa5f412837c9ec094e350b81

But we really need feedback from @ekalosak and @Phreedom!

Please accept the licenses as described in https://github.com/NixOS/nixos-homepage/pull/260#issuecomment-483310698!

**Comment by Phreedom** [comment]
I hereby license my contributions to this repository under the following licenses:

    Creative Commons Attribution Share Alike 4.0 International for website content
    MIT for the software (including sample code)

**Comment by silverhook** [comment]
I agree with @edolstra that this might be the case for those two commits.

**Comment by davidak** [comment]
Hello @follower, @sanmai-NL, @ghuntley,

thanks for your recent contributions!

We are about to properly license this repository.

Can you please accept the licenses as described in https://github.com/NixOS/nixos-homepage/pull/260#issuecomment-483310698!

**Comment by sanmai-NL** [comment]
I hereby license my contributions to this repository under the following licenses:

Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)

**Comment by grahamc** [comment]
bump on  @follower, @ghuntley, @ekalosak

**Comment by ghuntley** [comment]
I consent for you to use my work, do what you will with it, including relicensing as need be.  I don't care about particulars re: MIT or CC-BY-SA related to website changes.

**Comment by ghuntley** [comment]
I hereby license my contributions to this repository under the following licenses:

```
Creative Commons Attribution Share Alike 4.0 International for website content
MIT for the software (including sample code)
<any other license>
```

**Comment by davidak** [comment]
Still 2 missing: @follower, @ekalosak

Please see https://github.com/NixOS/nixos-homepage/pull/260#issuecomment-503153209

**Comment by zimbatm** [comment]
commits from @follower:
* 9ca1d11f682734b0aecd01ceb2889eaf731d466d
commits from @ekalosak:
* 3587643ec75a3544e7ded6b922c1ab7aab39e631

It's probably safe to go ahead and resolve potential licensing issues by replacing their work if they are unhappy.

**Comment by davidak** [comment]
@zimbatm ok. then let's finally merge it :rocket:

**Comment by grahamc** [comment]
:tada:

**Comment by grahamc** [comment]
If @ekalosak would rather not license his contribution this way, we can address it.

**Pull request merged**